plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -236,6 +241,56 @@ private List<SchemaTableName> listTablesInRemoteSchema(BigQueryClient client, St
         return tableNames.build();
     }
 
+    @Override
+    public Iterator<RelationCommentMetadata> streamRelationComments(ConnectorSession session, Optional<String> schemaName, UnaryOperator<Set<SchemaTableName>> relationFilter)
+    {
+        BigQueryClient client = bigQueryClientFactory.create(session);
+        List<String> schemaNames = schemaName.map(name -> {
+            DatasetId localDatasetId = client.toDatasetId(name);
+            String remoteSchemaName = getRemoteSchemaName(client, localDatasetId.getProject(), localDatasetId.getDataset());
+            return List.of(remoteSchemaName);
+        }).orElseGet(() -> listSchemaNames(session));
+        Map<SchemaTableName, RelationCommentMetadata> resultsByName = schemaNames.stream()
+                .flatMap(schema -> {
+                    try {
+                        return listRelationCommentMetadata(session, client, schema).stream();
+                    }
+                    catch (BigQueryException e) {
+                        if (e.getCode() == 404) {
+                            log.debug("Dataset disappeared during listing operation: %s", schema);
+                            return Stream.empty();
+                        }
+                        throw new TrinoException(BIGQUERY_LISTING_TABLE_ERROR, "Failed to retrieve tables from BigQuery", e);
+                    }
+                })
+                .collect(toImmutableMap(RelationCommentMetadata::name, Functions.identity()));
+        return relationFilter.apply(resultsByName.keySet()).stream()
+                .map(resultsByName::get)
+                .iterator();
+    }
+
+    private List<RelationCommentMetadata> listRelationCommentMetadata(ConnectorSession session, BigQueryClient client, String schemaName)
+    {
+        TableResult result = client.executeQuery(session, """
+                SELECT tbls.table_schema, tbls.table_name, options.option_value
+                FROM %1$s.`INFORMATION_SCHEMA`.`TABLES` tbls
+                LEFT JOIN %1$s.`INFORMATION_SCHEMA`.`TABLE_OPTIONS` options
+                ON tbls.table_schema = options.table_schema AND tbls.table_name = options.table_name AND options.option_name = 'description'
+                """.formatted(quote(schemaName)));
+        return result.streamValues()
+                .map(row -> {
+                    Optional<String> comment = row.get(2).isNull() ? Optional.empty() : Optional.of(unquote(row.get(2).getStringValue()));
+                    return new RelationCommentMetadata(new SchemaTableName(row.get(0).getStringValue(), row.get(1).getStringValue()), false, comment);
+                })
+                .collect(toImmutableList());
+    }
+
+    private static String unquote(String quoted)
```

Comment:
Note: Special characters on table comments are tested in `BaseConnectorTest#testCommentTableSpecialCharacter()`